### PR TITLE
secrets/ssh: allow algorithm_signer to use the key's default algo

### PR DESCRIPTION
**Pull request by calvn**
This PR allows certificates to be signed using the signer's default algorithm if none is specified in the role's `algorithm_signer` parameter.

Related to https://github.com/hashicorp/vault/pull/9096 
Fixes https://github.com/hashicorp/vault/issues/9739
Closes https://github.com/hashicorp/vault/issues/8414

### CODE REVIEW

**Comment by ncabatoff** [comment]
Can you add a case to TestSSHBackend_CA please?

**Comment by calvn** [comment]
Yes, I'm in the process of adding tests. Opened it in draft mode for early review.

File: builtin/logical/ssh/backend_test.go [hunk 1]
```diff
@@ -668,29 +680,33 @@ func TestSSHBackend_CredsForZeroAddressRoles_dynamic(t *testing.T) {
...
+		caPrivateKey string
+		algoSigner   string
+		expectError  bool
 	}{
-		{"defaultSignerSSHDSupport", dockerImageTagSupportsRSA, "", false},
-		{"rsaSignerSSHDSupport", dockerImageTagSupportsRSA, ssh.SigAlgoRSA, false},
-		{"rsa2SignerSSHDSupport", dockerImageTagSupportsRSA, ssh.SigAlgoRSASHA2256, false},
-		{"rsa2SignerNoSSHDSupport", dockerImageTagSupportsNoRSA, ssh.SigAlgoRSASHA2256, false},
-		{"defaultSignerNoSSHDSupport", dockerImageTagSupportsNoRSA, "", true},
+		{"RSAKey/EmptyAlgoSigner/ImageSupportsRSA1", dockerImageTagSupportsRSA1, testCAPublicKey, testCAPrivateKey, "", false},
```
Comment (ncabatoff) [inline builtin/logical/ssh/backend_test.go:690]: Think you might want to go multi-line here for readability.

Comment (ncabatoff) [reply builtin/logical/ssh/backend_test.go:690]: Also, mind replacing these slashes with dashes or something?  It makes them look like sub-sub-sub tests, and might complicate future efforts to list tests and schedule them across CI executors.

Comment (calvn) [reply builtin/logical/ssh/backend_test.go:690]: Yep, I'll address both suggestions.

**Pull request merged**